ext/LegendHDF5IOLegendDataManagementExt.jl

```diff
@@ -53,27 +53,25 @@ function LegendHDF5IO.LH5Array(ds::LegendHDF5IO.HDF5.Dataset, ::Type{<:AbstractA
 end
 
 function __init__()
-    LegendHDF5IO._datatype_dict["expsetup"] = ExpSetup
-    LegendHDF5IO._datatype_dict["datatier"] = DataTier
-    LegendHDF5IO._datatype_dict["datapartition"] = DataPartition
-    LegendHDF5IO._datatype_dict["dataperiod"] = DataPeriod
-    LegendHDF5IO._datatype_dict["datarun"] = DataRun
-    LegendHDF5IO._datatype_dict["datacategory"] = DataCategory
-    LegendHDF5IO._datatype_dict["timestamp"] = Timestamp
-    LegendHDF5IO._datatype_dict["filekey"] = FileKey
-    LegendHDF5IO._datatype_dict["channelid"] = ChannelId
-    LegendHDF5IO._datatype_dict["detectorid"] = DetectorId
 
-    dataselector_bytypes[ExpSetup] = "expsetup"
-    dataselector_bytypes[DataTier] = "datatier"
-    dataselector_bytypes[DataPartition] = "datapartition"
-    dataselector_bytypes[DataPeriod] = "dataperiod"
-    dataselector_bytypes[DataRun] = "datarun"
-    dataselector_bytypes[DataCategory] = "datacategory"
-    dataselector_bytypes[Timestamp] = "timestamp"
-    dataselector_bytypes[FileKey] = "filekey"
-    dataselector_bytypes[ChannelId] = "channelid"
-    dataselector_bytypes[DetectorId] = "detectorid"
+    function extend_datatype_dict(type, key)
```

Comment:
Explicitly require types  of the arguments here:
```julia
function extend_datatype_dict(type::Type, key::String)
```
to avoid conflicts with the types defined in line 12 where 
```julia
const dataselector_bytypes = Dict{Type, String}()
```